File: back-end/PointStorePA3/PointStorePA3-ejb/src/main/java/br/com/unifor/PA3/entitys/Usuario.java
package br.com.unifor.PA3.entitys;

import java.io.Serializable;
import java.util.Collection;

import javax.ejb.Stateless;
import javax.persistence.Column;
import javax.persistence.Entity;
import javax.persistence.GeneratedValue;
import javax.persistence.GenerationType;
import javax.persistence.Id;

@Entity
public class Usuario implements Serializable {

	private static final long serialVersionUID = 7578849729765946537L;
	
	@Id
	@Column(name="id_user")
	@GeneratedValue(strategy = GenerationType.IDENTITY)
	private Long id;
	private String nome;
	private String sobrenome;
	private String email;
	private String login;
	private String senha;
	private String cpf;
	private Double credito;
	private Integer vendas;


	public Long getId() {
		return id;
	}
	public void setId(Long id) {
		this.id = id;
	}
	public String getNome() {
		return nome;
	}
	public void setNome(String nome) {
		this.nome = nome;
	}
	public String getSobrenome() {
		return sobrenome;
	}
	public void setSobrenome(String sobrenome) {
		this.sobrenome = sobrenome;
	}
	public String getEmail() {
		return email;
	}
	public void setEmail(String email) {
		this.email = email;
	}
	public String getLogin() {
		return login;
	}
	public void setLogin(String login) {
		this.login = login;
	}
	public String getSenha() {
		return senha;
	}
	public void setSenha(String senha) {
		this.senha = senha;
	}
	public String getCpf() {
		return cpf;
	}
	public void setCpf(String cpf) {
		this.cpf = cpf;
	}
	public Double getCredito() {
		return credito;
	}
	public void setCredito(Double credito) {
		this.credito = credito;
	}
	public Integer getVendas() {
		return vendas;
	}
	public void setVendas(Integer vendas) {
		this.vendas = vendas;
	}
	
	
}

File: back-end/src/pointstore/entity/Usuario.php
<?php

namespace pointstore\entity;


/**
 * @Entity
 * @Table(name="usuario")
 */
class Usuario{

	/**
	*	@var integer @Id
	*      @Column(type="integer")
	*      @GeneratedValue(strategy="AUTO")
	*/
	private $id;

	/**
	*
	* @var string @Column(type="string", length=255)
	*/
	private $nome;
	
	/**
	*
	* @var string @Column(type="string", length=255)
	*/
	private $sobrenome;
	
	/**
	*
	* @var string @Column(type="string", length=255)
	*/
	private $cpf;
	
	/**
	*
	* @var integer @Column(type="integer", length=255)
	*/
	private $credito;
	
	/**
	*
	* @var string @Column(type="string", length=255)
	*/
	private $email;
	
	/**
	*
	* @var string @Column(type="string", length=255)
	*/
	private $login;
	
	/**
	*
	* @var string @Column(type="string", length=255)
	*/
	private $senha;

	/**
	 * @OneToMany(targetEntity="\pointstore\entity\MeusPontos", mappedBy="IdUsuario", cascade={"persist", "remove", "merge"}, orphanRemoval=true)
	 */
	private $meusPontos;

	/**
	 * @OneToMany(targetEntity="\pointstore\entity\Venda", mappedBy="idUsuarioVendedor", cascade={"persist", "remove", "merge"}, orphanRemoval=true)
	 */
	private $idUsuarioVendedor;

	/**
	 * @OneToMany(targetEntity="\pointstore\entity\Venda", mappedBy="idUsuarioComprador", cascade={"persist", "remove", "merge"}, orphanRemoval=true)
	 */
	private $idUsuarioComprador;


	
	public function getId(){
		return $this->id;
	}

	public function setId($id){
		$this->id = $id;
	}

	public function getNome(){
		return $this->nome;
	}

	public function setNome($nome){
		$this->nome = $nome;
	}

	public function getSobrenome(){
		return $this->sobrenome;
	}

	public function setSobrenome($sobrenome){
		$this->sobrenome = $sobrenome;
	}

	public function getCpf(){
		return $this->cpf;
	}

	public function setCpf($cpf){
		$this->cpf = $cpf;
	}

	public function getCredito(){
		return $this->credito;
	}

	public function setCredito($credito){
		$this->credito = $credito;
	}

	public function getEmail(){
		return $this->email;
	}

	public function setEmail($email){
		$this->email = $email;
	}

	public function getLogin(){
		return $this->login;
	}

	public function setLogin($login){
		$this->login = $login;
	}

	public function getSenha(){
		return $this->senha;
	}

	public function setSenha($senha){
		$this->senha = $senha;
	}

	public function getMeusPontos(){
		return $this->meusPontos;
	}

	public function setMeusPontos($meusPontos){
		$this->meusPontos = $meusPontos;
	}

	public function getIdUsuarioVendedor(){
		return $this->idUsuarioVendedor;
	}

	public function setIdUsuarioVendedor($idUsuarioVendedor){
		$this->idUsuarioVendedor = $idUsuarioVendedor;
	}

	public function getIdUsuarioComprador(){
		return $this->idUsuarioComprador;
	}

	public function setIdUsuarioComprador($idUsuarioComprador){
		$this->idUsuarioComprador = $idUsuarioComprador;
	}

	public function __construct(){}

	public static function construct($array){
		$obj = new Usuario();
		$obj->setNome( $array['nome']);
		$obj->setSobrenome( $array['sobrenome']);
		$obj->setEmail( $array['email']);
		$obj->setLogin( $array['login']);
		$obj->setSenha( $array['senha']);
		return $obj;
	}

	public function toString(){
 		return "  [id:" .$this->id. "]  [nome:" .$this->nome. "]  [sobrenome:" .$this->sobrenome. "]  
 		[cpf:" .$this->cpf. "]  [credito:" .$this->credito. "]  [email:" .$this->email. "]  [login:" .$this->login. "]  
 		[senha:" .$this->senha. "]	[vendas:" .$this->vendas. "]  ";
	}

 	public function toArray(){
   		return [
  				"id"=>$this->id,
  				"nome"=>$this->nome,
  				"sobrenome"=>$this->sobrenome,
  				"cpf"=>$this->cpf,
  				"credito"=>$this->credito,
   				"email"=>$this->email,
   				"login"=>$this->login,
   				"senha"=>$this->senha,
   				"vendas"=>$this->vendas];
 	}
	
}
